contracts/lib/LibPrices.sol

```diff
@@ -92,7 +101,7 @@ library Prices {
 
         for (uint256 i = 0; i < 8; i++) {
             uint256 currTimeWeight = 8 - i;
-            uint256 j = 8 - i;
+            uint256 j = hour < i ? 23 : hour - i;
```

Comment:
If, for example, `hour == 2`, then when i is from 3 -> 7, `j` is going to be 23 for all of those iterations.

Is this OK?

Reply:
@BenjaminPatch good catch. This is actually a bug in the old implementation that nobody caught 👀 

It should be continue looping towards 0 rather than repeating 23. Will fix